### PR TITLE
[Refactor #1] -  Better way to validate, imports and avoid duplicated code.

### PR DESCRIPTION
**Pull request by Jhooomn**
- There was a little code refactor dedicated to avoid :
 * null or not null validation, use objets class
 * bad imports with overload of resources
 * there was code which were not being used so --> remove
 * duplication of code is very common
}

### CODE REVIEW

File: core/src/main/java/com/usthe/sureness/processor/support/DigestProcessor.java [hunk 1]
```diff
@@ -66,11 +72,8 @@ public Subject authenticated(Subject var) throws SurenessAuthenticationException
             throw new UnknownAccountException("do not exist the account: " + appId);
         }
         DigestSubject digestSubject = (DigestSubject) var;
```
Comment (tomsun28) [inline core/src/main/java/com/usthe/sureness/processor/support/DigestProcessor.java:74]:
```suggestion
        DigestSubject digestSubject = (DigestSubject) var;
        //A1 = MD5("username:realm:password");
```

File: core/src/main/java/com/usthe/sureness/processor/support/DigestProcessor.java [hunk 2]
```diff
@@ -66,11 +72,8 @@ public Subject authenticated(Subject var) throws SurenessAuthenticationException
...
-        //A1 = MD5("username:realm:password");
         String a1 = calcDigest(appId, digestSubject.getRealm(), account.getPassword());
```
Comment (tomsun28) [inline core/src/main/java/com/usthe/sureness/processor/support/DigestProcessor.java:75]:
```suggestion
        String a1 = calcDigest(appId, digestSubject.getRealm(), account.getPassword());
        //A2 = MD5("httpMethod:uri");
```

File: core/src/main/java/com/usthe/sureness/processor/support/DigestProcessor.java [hunk 3]
```diff
@@ -66,11 +72,8 @@ public Subject authenticated(Subject var) throws SurenessAuthenticationException
...
-        //A2 = MD5("httpMethod:uri");
         String a2 = calcDigest(digestSubject.getHttpMethod(), digestSubject.getUri());
```
Comment (tomsun28) [inline core/src/main/java/com/usthe/sureness/processor/support/DigestProcessor.java:76]:
```suggestion
        String a2 = calcDigest(digestSubject.getHttpMethod(), digestSubject.getUri());
        //response = MD5("A1:nonce:nc:cNonce:qop:A2");
```

File: core/src/main/java/com/usthe/sureness/processor/support/DigestProcessor.java [hunk 4]
```diff
@@ -82,19 +85,26 @@ public Subject authenticated(Subject var) throws SurenessAuthenticationException
         if (account.isExcessiveAttempts()) {
             throw new ExcessiveAttemptsException("account is disable due to many time authenticated, try later");
         }
-        // attention: need to set subject own roles from account
         var.setOwnRoles(account.getOwnRoles());
```
Comment (tomsun28) [inline core/src/main/java/com/usthe/sureness/processor/support/DigestProcessor.java:88]:
```suggestion
        // attention: need to set subject own roles from account
        var.setOwnRoles(account.getOwnRoles());
```

File: core/src/main/java/com/usthe/sureness/processor/support/DigestProcessor.java [hunk 5]
```diff
@@ -117,11 +127,5 @@ public void setAccountProvider(SurenessAccountProvider provider) {
         this.accountProvider = provider;
     }
```
Comment (tomsun28) [inline core/src/main/java/com/usthe/sureness/processor/support/DigestProcessor.java:129]:
```suggestion
    public static void setRealm(String realm) {
        DigestProcessor.realm = realm;
    }

    public static void setQop(String qop) {
        DigestProcessor.qop = qop;
    }
```

**Review by tomsun28 — COMMENTED** [review]
Thanks! Most of them look good to me. Here some review suggest. By the way, are these automatically repaired by tool? cool.

File: core/src/main/java/com/usthe/sureness/matcher/DefaultPathRoleMatcher.java [hunk 6]
```diff
@@ -118,22 +99,18 @@ public void rebuildTree() {
...
-        return exclude != null && exclude.equals(EXCLUDE_ROLE);
+        return nonNull(exclude) && exclude.equals(EXCLUDE_ROLE);
     }
 
     private void checkComponentInit() {
-        if (pathTreeProviderList == null) {
+        if (isNull(pathTreeProviderList)) {
             throw new SurenessNoInitException("DefaultPathRoleMatcher init error : component init not complete");
         }
     }
```
[the remark this answers is not in the document]
Comment (Jhooomn) [reply core/src/main/java/com/usthe/sureness/matcher/DefaultPathRoleMatcher.java]: Oh! It's already back!

**Comment by Jhooomn** [comment]
@tomsun28 updated!

**Comment by tomsun28** [comment]
LGTM thanks!

**Pull request merged**